File: ios/Methods/UserInteraction/CancelMethodHandler.swift
//
// Copyright © 2023 Nevis Security AG. All rights reserved.
//

import NevisMobileAuthentication

struct CancelMethodHandler: MethodHandler {
	func execute(using _: MobileAuthenticationClient, with message: ChannelInMessage) throws {
		let message: CancelMessage = try validate(message: message)
		OperationCache.shared.cancel(by: message.operationId)
		MethodChannelHandler.shared.resolve(method: .cancel, message: nil)
	}
}

File: android/src/main/java/ch/nevis/mobile/authentication/sdk/react/methods/userInteraction/CancelMethodHandler.kt
/*
 * Copyright © 2023 Nevis Security AG. All rights reserved.
 */

package ch.nevis.mobile.authentication.sdk.react.methods.userInteraction

import ch.nevis.mobile.authentication.sdk.react.MethodChannelHandler
import ch.nevis.mobile.authentication.sdk.react.cache.OperationCache
import ch.nevis.mobile.authentication.sdk.react.methods.MethodHandler
import ch.nevis.mobile.authentication.sdk.react.model.ReactMethodName
import ch.nevis.mobile.authentication.sdk.react.model.message.incoming.CancelMessage
import ch.nevis.mobile.authentication.sdk.react.model.message.incoming.ChannelInMessage
import ch.nevis.mobile.sdk.api.MobileAuthenticationClient

class CancelMethodHandler : MethodHandler() {
    override fun execute(
        client: MobileAuthenticationClient,
        channelInMessage: ChannelInMessage,
    ) {
        val message = validate<CancelMessage>(channelInMessage)
        OperationCache.cancel(message.operationId)
        MethodChannelHandler.resolve(ReactMethodName.CANCEL, null)
    }
}
